src/consentHandler.js

```diff
@@ -0,0 +1,68 @@
+
+export class ConsentHandler {
+  #enabled;
+  #data;
+  #promise;
+  #resolve;
+  #ready;
+
+  constructor() {
+    this.reset();
+  }
+
+  /**
+   * reset this handler (mainly for tests)
+   */
+  reset() {
+    this.#promise = new Promise((resolve) => {
+      this.#resolve = (data) => {
+        this.#ready = true;
+        this.#data = data;
+        resolve(data);
+      };
+    });
+    this.#enabled = false;
+    this.#data = null;
+    this.#ready = false;
+  }
+
+  /**
+   * Enable this consent handler. This should be called by the relevant consent management module
+   * on initialization.
+   */
+  enable() {
+    this.#enabled = true;
+  }
+
+  /**
+   * @returns {boolean} true if the related consent management module is enabled.
+   */
+  get enabled() {
+    return this.#enabled;
+  }
+
+  /**
+   * @returns {boolean} true if consent data has been resolved (it may be `null` if the resolution failed).
+   */
+  get ready() {
+    return this.#ready;
+  }
+
+  /**
+   * @returns a promise than resolves to the consent data, or null if no consent data is available
+   */
+  get promise() {
+    if (!this.#enabled) {
+      this.#resolve(null);
```

Comment:
Will rejecting the promise suit better here?

Reply:
I don't think so (but I'm not that sure). My reasoning is that unless you make it so that `data` is *never* `null` when the promise resolves, it's just asking the caller to add some complexity, because they will have to check for `null` and treat it as an error anyways. I am not against making every `null` (including, for example, USP timeouts) a rejection, but it also makes sense to me that you would want to have exactly what `getConsentData` returns, just asynchronously.

If someone is interested in whether the module is installed, they could look at `.enabled`.